syncer/syncer_test.go

```diff
@@ -570,3 +633,45 @@ func TestSync_AlsoSyncProcessedLayer(t *testing.T) {
 	// but last synced is updated
 	require.Equal(t, lyr, ts.syncer.getLastSyncedLayer())
 }
+
+func TestSyncer_setSyncedTwice_NoError(t *testing.T) {
+	ts := newSyncerWithoutSyncTimer(t)
+
+	sync := ts.syncer.RegisterForSynced(context.TODO())
+	select {
+	case <-sync:
+		require.Fail(t, "should not have reached synced state yet")
+	case <-time.After(100 * time.Millisecond):
+	}
+
+	ts.syncer.setSyncState(context.TODO(), synced)
+
+	select {
+	case <-sync:
+	case <-time.After(1 * time.Second):
+		require.Fail(t, "should have reached synced state")
+	}
+
+	require.NotPanics(t, func() { ts.syncer.setSyncState(context.TODO(), synced) })
```

Comment:
Out of curiosity - what's the purpose of `require.NotPanics`? The test should fail if the code panics anyway, shouldn't it? What is the point?

Reply:
The called method used to panic before a bug was fixed with this PR. I wrote this test to explicitly check if the bug was fixed.

If I remove the `require.NotPanics` I would need to add a comment of the form:

```golang
// call setSyncState to check that it doesn't throw
ts.syncer.setSyncState(context.TODO(), synced)
```

Also if it does throw, the require prints an error message and continue with the next test. If not recovered the panic will stop the execution of the remaining tests (at least for the package).